docs/manual/scala/guide/build/ScalaComponents.md

```diff
@@ -0,0 +1,68 @@
+
+### A brief intro to Cake Pattern
+
+The cake pattern is a design pattern for Dependency Injection in scala where an application (the cake) is built as by mixing in several traits (the cake slices). Each trait may declare unimplemented `def` methods stating what it requires to run. Each trait may provide implementations the final application will need. To present an example, many Lagom Components (or cake layers) declare a need for a `ServiceLocator` and as a developer you are required to provide one when using one of those Components. If you don't fulfill all the declared requirements for the cake slices you mix into your cake, it will be incomplete and compilation will fail.
+
+Let's see an example with actual Lagom Scala API code. Imagine you have a `HelloService` (and a `HelloServiceImpl`) and you are building your Application cake. You could prepare:
+
+@[lagom-application](code/ServiceImplementation.scala)
+
+That is a simple enough Application but if you pay close attention you'll notice it's is an `abstract class`. The `HelloApplication` builds all the basics for the application to be run but it's not complete. It extends [`LagomApplication`](api/com/lightbend/lagom/scaladsl/server/LagomApplication.html) so it needs a `serviceLocator: ServiceLocator`, a `lagomService: LagomServer` and a `wcClient: WSClient`. We provide the `wcClient: WSClient` mixing in `AhcWSComponents` and we provide `lagomService: LagomServer` programmatically. We still need a `serviceLocator: ServiceLocator` but we need to specify a different [[Service Locator|ServiceLocator]] for each environment we want our app to run in (development environment or production environment):
+
+@[lagom-loader](code/ServiceImplementation.scala)
+
+In the `ApplicationLoader` depending on the runtime environment we will complete the cake in a slightly different way. See how the `ApplicationLoader` defines two methods: `loadDevMode()` and `load()` so you can mix-in different ServiceLocators. In `loadDevMode()` we mix in `LagomDevModeComponents` which provide a Service Locator for Lagom's development mode environment and will also add a feature into our service to self register when the service boots up.
+
+This is a brief introduction to Lagom Application cake, there's more information in [[Wiring together a Lagom application|ServiceImplementation#Wiring-together-a-Lagom-application]].
+
+## Components
+
+This is a list of available `Components` you may use to build your application cake. This list contains Components provided by Lagom, Play and other projects or libraries. You may develop your own Components and use those instead of the provided on this list as long as the contracts are fulfilled.
+
+|  Service Components  ||
```

Comment:
The table headers should be broken out of the table and rendered as markdown headings, so that they have individual anchors, and also because they logically don't form part of the table data.